tiledb/sm/fragment/single_fragment_info.h

```diff
@@ -119,10 +119,8 @@ class SingleFragmentInfo {
   }
 
   /** Dumps the single fragment info in ASCII format in the selected output. */
-  void dump(const std::vector<Datatype>& dim_types, FILE* out) const {
-    if (out == nullptr)
-      out = stdout;
-
+  std::string dump_single_fragment_info(
+      const std::vector<Datatype>& dim_types) const {
```

Comment:
I think this should be an `operator<<` as well.

Reply:
The problem is the `dim_types` parameter of `dump_single_fragment_info` function.
Any ideas on how to overcome it and make `operator<<` work?

Reply:
In this case I would suggest you to change the method's signature to this:

```suggestion
  void dump_single_fragment_info(
      std::ostream& os, const std::vector<Datatype>& dim_types) const {
```

This way we will avoid the intermediate string allocations and bring ourselves as close to the pattern as we can.